homeassistant/components/jewish_calendar/binary_sensor.py

```diff
@@ -34,9 +34,6 @@ async def async_setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the Jewish Calendar binary sensor devices."""
-    if discovery_info is None:
```

Comment:
This is needed to protect against incorrect configuration such as:
```yaml
binary_sensor:
  - platform: jewish_calendar
 ```

Removing it may cause an exception for the above configuration:
```
2022-04-27 22:29:13 ERROR (MainThread) [homeassistant.components.binary_sensor] Error while setting up jewish_calendar platform for binary_sensor
Traceback (most recent call last):
  File "/workspaces/core/homeassistant/helpers/entity_platform.py", line 249, in _async_setup_platform
    await asyncio.shield(task)
  File "/workspaces/core/homeassistant/components/jewish_calendar/binary_sensor.py", line 40, in async_setup_platform
    async_add_entities([JewishCalendarBinarySensor(hass.data[DOMAIN], BINARY_SENSORS)])
KeyError: 'jewish_calendar'
```